### PR TITLE
[HOPSWORKS-2750] Git logs

### PR DESCRIPTION
**Pull request by gibchikafa**

### CODE REVIEW

**Review by jimdowling — COMMENTED** [review]
Is the purpose of this PR to save all the logs from git operations?
I assume yes.

File: recipes/_filebeat-git.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
+file "#{node['filebeat']['base_dir']}/filebeat.xml" do
+  action :delete
+end
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:3]: Ok, the other recipes do this. I don't know why, but i guess the operation is idempotent.

File: recipes/_filebeat-git.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+git_owner = ""
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:5]: Change these lines to 
  git_owner = "glassfish"
  git_group = "glassfish"
  
  Then you can remove the else clause in the statement below.

File: recipes/_filebeat-git.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+git_group = ""
+if node.attribute?("hopsworks") && node['hopsworks'].attribute?("user")
+  git_owner = node['hopsworks']['user']
+  git_group = node['hopsworks']['user']
+else
+  git_owner = "glassfish"
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:11]: Remove this else clause, see comment above.

File: recipes/_filebeat-git.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  git_group = "glassfish"
+end
+
+#Add glassfish user to elastic group
+group node['elastic']['user'] do
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:16]: This should be
node['elastic']['group']
right?

File: recipes/_filebeat-git.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  action :modify
+  members [git_owner]
+  append true
+  not_if { node['install']['external_users'].casecmp("true") == 0 }
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:20]: nice. need this.

File: recipes/_filebeat-git.rb [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+end
+
+log_glob = "#{node['install']['dir']}/git/*/*/*/*/logs/*.log"
+if node.attribute?("git") && node['git'].attribute?("base_dir")
+  log_glob = "#{node['git']['base_dir']}/*/*/*/*/logs/*.log"
+end
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:26]: I trust this works, have no idea.

File: recipes/_filebeat-git.rb [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+logstash_fqdn = consul_helper.get_service_fqdn("logstash")
+logstash_endpoint = "#{logstash_fqdn}:#{node['logstash']['beats']['git_port']}"
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:29]: looks good.

File: recipes/_filebeat-git.rb [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  group git_group
+  mode 0655
+  variables({
+              :paths => [log_glob],
+              :multiline => false,
+              :fields => false,
+              :logstash_endpoint => logstash_endpoint,
+              :log_name => "git"
+            })
+end
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:43]: ok

File: recipes/_filebeat-git.rb [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+template "#{node['filebeat']['base_dir']}/bin/start-filebeat-git.sh" do
+  source "start-filebeat.sh.erb"
+  user git_owner
+  group git_group
+  mode 0750
+  variables({
+              :pid => "#{node['filebeat']['pid_dir']}/filebeat-git.pid",
+              :config_file => "filebeat-git.yml"
+            })
+end
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:54]: ok

File: recipes/_filebeat-git.rb [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  service "ELK"
+  log_file "#{node['filebeat']['base_dir']}/log/git"
+end
+
+
+if conda_helpers.is_upgrade
+  kagent_config "#{service_name}" do
+    action :systemd_reload
+  end
+end
```
Comment (jimdowling) [inline recipes/_filebeat-git.rb:115]: All of this is cut-and-paste from the other filebeat services. ok.
At some point we should just have 1 filebeat-service.rb file that is parameterized  by jupyter, git, serving.

File: recipes/_logstash.rb [hunk 11]
```diff
@@ -53,6 +53,17 @@
...
+  source "git.conf.erb"
+  owner node['hopslog']['user']
+  group node['hopslog']['group']
+  mode 0655
+  variables({
+              :elastic_addr => elastic_addrs,
+              :hops_ca => hops_ca
+            })
+end
+
```
Comment (jimdowling) [inline recipes/_logstash.rb:66]: ok

File: templates/default/consul/logstash-consul.hcl.erb [hunk 12]
```diff
@@ -62,5 +62,17 @@ services = [
...
+        name = "logstash"
+        tags = [
+            "git"
+        ]
+        port = <%= node['logstash']['beats']['git_port'] %>
+        check = {
+            id = "logstash-beats-git-check"
+            alias_service = "logstash/http"
+        }
     }
```
Comment (jimdowling) [inline templates/default/consul/logstash-consul.hcl.erb:77]: ok

**Review by jimdowling — CHANGES_REQUESTED** [review]
Tiny fix.

**Comment by gibchikafa** [comment]
Replaced with https://github.com/logicalclocks/hopslog-chef/pull/93

**Pull request closed without merging**